File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int x = 0;
        int y = 1;
        while(y<nums.size())
        {
            if(nums[y]>nums[x])
            {
                x++;
                nums[x] = nums[y];
            }
                y++;
        }
        return x+1;
    }
};

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int prev = Integer.MIN_VALUE;
        int i = 0;
        int j = 0;
        int count = 0;
        while(j<nums.length)
        {
            if(nums[j]>prev)
            {
                prev = nums[j];
                int temp = nums[j];
                nums[j] = nums[i];
                nums[i] = temp;
                i++;
                j++;
                count++;
            }
            else
            {
                j++;
            }
        }
        return count;
    }
}